public/app/plugins/panel/logs/LogsPanelEditor.tsx

```diff
@@ -25,8 +55,30 @@ export class LogsPanelEditor extends PureComponent<PanelEditorProps<Options>> {
     onOptionsChange({ ...options, sortOrder: item.value });
   };
 
+  onDataLinksChange = (dataLinks: DataLink[]) => {
+    const { options, onOptionsChange } = this.props;
+    onOptionsChange({ ...options, dataLinks });
+  };
+
+  getSuggestions = async (): Promise<VariableSuggestion[]> => {
+    const { datasource } = this.props;
+    const fields = await datasource.getDerivedFields({});
+    const suggestions: VariableSuggestion[] = [];
+    for (const key of Object.keys(fields)) {
+      const prefix = key;
+      for (const key2 of Object.keys(fields[key].value)) {
+        suggestions.push({
+          value: `${prefix}.${key2}`,
+          label: key2,
+          origin: VariableOrigin.Field,
+        });
+      }
+    }
+    return suggestions;
+  };
```

Comment:
This should use the getDataLinksVariableSuggestions that take DataFrame[] (it adds dashboard template variables)
https://github.com/grafana/grafana/blob/master/public/app/features/panel/panellinks/link_srv.ts#L100

In my mind, getDerivedFields is an internal loki thing (as much as possible). it just adds DataFrame fields to the normal response